tests/k8s/k8s-gsg-test.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+NAMESPACE="kube-system" 
+
+source "../helpers.bash"
+source /home/vagrant/.profile
+function cleanup {
+	kubectl delete -f https://raw.githubusercontent.com/cilium/cilium/master/examples/minikube/l3_l4_l7_policy.yaml
+	kubectl delete -f https://raw.githubusercontent.com/cilium/cilium/master/examples/minikube/l3_l4_policy.yaml
+	kubectl delete -f https://raw.githubusercontent.com/cilium/cilium/master/examples/minikube/demo.yaml
+	kubectl delete -f cilium-ds.yaml
+	kubectl delete -f https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/rbac.yaml
+}
+
+trap cleanup exit
+
+echo "KUBECONFIG: $KUBECONFIG"
+
+cleanup
+
+until [ "$(kubectl get cs | grep -v "STATUS" | grep -c "Healthy")" -eq "3" ]; do 
+	echo "---- Waiting for cluster to get into a good state ----"
+	sleep 5
+done
+
+
+echo "----- adding RBAC for Cilium -----"
+kubectl create -f https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/rbac.yaml
+
+echo "----- deploying Cilium Daemon Set onto cluster -----"
+wget https://raw.githubusercontent.com/cilium/cilium/master/examples/kubernetes/cilium-ds.yaml
+sed -i s/"\/var\/lib\/kubelet\/kubeconfig"/"\/etc\/kubernetes\/kubelet.conf"/g cilium-ds.yaml
+sed -i s/"cilium\/cilium:stable"/"localhost:5000\/cilium:build_test"/g cilium-ds.yaml
+kubectl apply -f cilium-ds.yaml
+
+until [ "$(kubectl get ds --namespace ${NAMESPACE} | grep -v 'READY' | awk '{ print $4}' | grep -c '1')" -eq "3" ]; do
+	echo "----- Waiting for Cilium to get into 'ready' state in Minikube cluster -----"
+	sleep 5
+done
+
+# Let Cilium spin up.
+sleep 15
+
+echo "----- deploying demo application onto cluster -----"
+kubectl create -f https://raw.githubusercontent.com/cilium/cilium/master/examples/minikube/demo.yaml
```

Comment:
ds state != cilium state. We would need to loop until `cilium status` reports OK eventually. Let's keep the sleep but open a github issue to improve this.